### ISSUE TITLE
Build fails with HDF5 1.13.2 and 1.13.3

### ISSUE THREAD

**Issue by thebaptiste** [OSGeo/gdal#6657]
Hello

Building gdal 3.5.3 from sources against hdf5 1.13.3 on Rockylinux 8 with gcc/g++ 8.5.0, I have the following error : 

```
In file included from hdf5dataset.cpp:36:
hdf5vfl.h:119:1: error: braces around scalar initializer for type ‘herr_t (*)(H5FD_t*, hid_t, hbool_t)’ {aka ‘int (*)(H5FD_t*, long int, bool)’}
 };
 ^
```

Is there an option to prevent this error ?

I have the same error with gdal 3.5.2. It was ok with hdf5 1.10, I did'nt try with hdf5 1.12

Regards

**Comment by rouault** [comment]
fix in https://github.com/OSGeo/gdal/pull/6660.

**Comment by thebaptiste** [comment]
Thanks.
Will you integrate this fix in a new release soon (may be 3.5.4) ?
Until then I will build against hdf5 1.13.1 or 1.12.2

**Comment by rouault** [comment]
> Will you integrate this fix in a new release soon (may be 3.5.4) ?

this will be in 3.6.1 in ~ 2 months. This is too late for 3.6.0 that will be released today
